metadata/metadata_test.go

```diff
@@ -335,3 +335,23 @@ func BenchmarkFromOutgoingContext(b *testing.B) {
 		FromOutgoingContext(ctx)
 	}
 }
+
+func BenchmarkFromIncomingContext(b *testing.B) {
+	md := Pairs(
+		"X-My-Header-1", "42",
+	)
+	ctx := NewIncomingContext(context.Background(), md)
+	for n := 0; n < b.N; n++ {
+		FromIncomingContext(ctx)
```

Comment:
Given that we are not using the return value from this function, does this benchmark even do anything useful? How can we be sure that the compiler does not optimize this statement out?

Reply:
Previous benchmarks in this test also don't use the return value. So, maybe we are good there.